src/feature_builder.hpp

```diff
@@ -109,44 +201,34 @@ struct feature_builder_visitor
                 }
             }
         }
-        if (valid) finalize(feature_builder);
+        if (valid) finalize(feature_builder, feature);
     }
-
-    void operator()(mapbox::geometry::polygon<coordinate_type> const& poly)
+    void apply_geometry_polygon(vtzero::feature const& feature)
     {
-        auto extent = mapbox::geometry::envelope(poly);
-        if (!boost::geometry::intersects(extent, bbox_)) return;
+        std::vector<detail::polygon_ring<CoordinateType>> rings;
+        vtzero::decode_polygon_geometry(feature.geometry(), detail::polygon_handler<CoordinateType>(rings, dx_, dy_, zoom_factor_));
+
         std::vector<mapbox::geometry::polygon<coordinate_type>> result;
```

Comment:
Like https://github.com/mapbox/vtcomposite/pull/56, I think this `result` either needs to be moved down into the `for (auto const& poly : mp)` loop, or cleared each loop. Right?